metaflow/plugins/parallel_decorator.py

```diff
@@ -122,8 +122,24 @@ def _local_multinode_control_task_step_func(flow, env_to_use, step_func, retry_c
     os.environ["MF_PARALLEL_NODE_INDEX"] = "0"
     step_func()
 
-    # join the subprocesses
-    for p in subprocesses:
-        p.wait()
+    # join the subprocesses and store logs
+    flow_datastore = flow._datastore.parent_datastore
+    errors = []
+    for task_id, p in subprocesses:
+        stdout, stderr = p.communicate()
```

Comment:
This may not be a concern but this has the potential to deadlock if too much output is produced on the stdout/stderr pipes AND the parallel processes actually need to all make forward progress (ie: they all have their own internal synchronization mechanism).

Reply:
I don't think it fixes it. Suppose two subprocesses A and B and somehow, A and B have a sync in them (for eg: A and B both need to sync at a barrier). Suppose further that A and B produce enough output to saturate the PIPEs. You would first `communicate` with A but that would never return because `communicate` waits for A to complete but that would never happen because B can't make progress because it is stuck on a full pipe and A can't terminate because it is stuck on waiting for B.